drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/editor/GuidedDecisionTableEditorPresenter.java

```diff
@@ -215,14 +227,71 @@ public void makeMenuBar() {
                 .build();
     }
 
+    protected Command getSaveAndRenameCommand() {
+
+        return saveAndRenameCommandBuilder
+                .addPathSupplier(getPathSupplier())
+                .addValidator(assetUpdateValidator)
+                .addRenameService(service)
+                .addMetadataSupplier(getMetadataSupplier())
+                .addContentSupplier(getContentSupplier())
+                .addIsDirtySupplier(getIsDirtySupplier())
+                .build();
+    }
+
+    private Supplier<Path> getPathSupplier() {
+        return () -> versionRecordManager.getPathToLatest();
+    }
+
+    Supplier<GuidedDecisionTable52> getContentSupplier() {
+        return () -> getActiveDocument().getModel();
+    }
+
+    Supplier<Metadata> getMetadataSupplier() {
+
+        return () -> {
+
+            final Overview overview = getActiveDocument().getOverview();
+
+            return overview.getMetadata();
+        };
+    }
+
+    Supplier<Boolean> getIsDirtySupplier() {
+
+        return () -> getAvailableDecisionTables().stream().anyMatch(dtPresenter -> {
```

Comment:
Is it not sufficient to simply use ```getActiveDocument()``` in the "single" Guided Decision Table editor and use this implementation in the "graph" editor?

Reply:
Good catch! Fixed.